### PR TITLE
Add search commands to rohdeschwarz

### PR DESCRIPTION
**Pull request by AndersBuhl**

### CODE REVIEW

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 1]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+        self._search_state = value
+        self._set_cache_valid()
+
+    def _get_search_source(self):
+        if not self._driver_operation_simulate:
+            value = self._ask("search:source?").lower()
+            if value[0:2] == 'ch':
+                self.search_source = 'channel' + value[-1]
+            elif value[0] == 'd':
+                self.search_source == 'digital' + value[-1]
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: does it return both the digits? 16 digital channels

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 2]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+        if hasattr(value, 'name'):
+            value = value.name
+        value = str(value)
+        if value not in self._channel_name + ['line']:
+            raise ivi.UnknownPhysicalNameException()
+        if not self._driver_operation_simulate:
+            if value[0:2] == 'ch':
+                scpi_string = 'ch' + value[-1]
+            elif value[0].lower() == 'd':
+                scpi_string = 'd' + value[-1]
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: same as above

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 3]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+            else:
+                self.search_source = value
+        self._set_cache_valid()
+        return self.search_source
+
+    def _set_search_source(self, value):
+        if hasattr(value, 'name'):
+            value = value.name
+        value = str(value)
+        if value not in self._channel_name + ['line']:
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: does it need value.lower() like below?

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 4]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+
+        self._search_source = value
+        self._set_cache_valid()
+
+    def _get_search_trigger_edge_level(self):
+        if not self._driver_operation_simulate:
+            self._search_trigger_edge_level = self._ask("search:trigger:edge:level?")
+
+        self._set_cache_valid()
+        return vaself._search_trigger_edge_levellue
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: vaself..

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 5]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+            self._write("search:source %s" % scpi_string)
+
+        self._search_source = value
+        self._set_cache_valid()
+
+    def _get_search_trigger_edge_level(self):
+        if not self._driver_operation_simulate:
+            self._search_trigger_edge_level = self._ask("search:trigger:edge:level?")
+
+        self._set_cache_valid()
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: why is set_cache_valid used in this case

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 6]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+            value = self._ask("search:edge:slope?").lower()
+            self._search_trigger_edge_slope = [k for k,v in SlopeMapping.items() if v==value][0]
+            self._set_cache_valid()
+        return self._search_trigger_edge_slope
+
+    def _set_search_trigger_edge_slope(self, value):
+        if value not in SlopeMapping:
+            raise ivi.ValueNotSupportedException()
+        if not self._driver_operation_simulate:
+            self._write("search:edge:slope %s" % SlopeMapping[value])
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: is it same as search.trigger:edge:slope

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 7]
```diff
@@ -1120,3 +1202,100 @@ def _set_reference_level_middle(self, value):
...
+        self._search_trigger_edge_slope = SlopeMapping[value]
+        self._set_cache_valid()
+
+    def _get_search_condition(self):
+        if not self._driver_operation_simulate:
+            value = self._ask("search:condition?")
+            self._search_condition = [k for k,v in SearchConditionMapping.items() if v==value][0]
```
Comment (Baijualex) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: is value in lower case

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 8]
```diff
@@ -83,6 +83,15 @@
         'positive': 'pos',
         'negative': 'neg',
         'either': 'eith'}
+SearchConditionMapping = {
```
Comment (JonasLang) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: I think these should be in the MeasurementFunctionMapping

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py [hunk 9]
```diff
@@ -181,6 +190,11 @@ def __init__(self, *args, **kwargs):
         self._trigger_type = 'edge'
         self._trigger_continuous = True
 
+        self._search_state = 'off'
+        self._search_condition = 'edge'
+        self._search_trigger_edge_slope = 'positive'
```
Comment (JonasLang) [inline ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: These should probably be under self._trigger_xyz

File: ivi/rohdeschwarz/rohdeschwarzBaseScope.py (hunk 8 above)
Comment (AndersBuhl) [reply ivi/rohdeschwarz/rohdeschwarzBaseScope.py]: Fixed

**Pull request merged**